cluster/evaluator/evaluator_ringmaster_wrapper.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+evaluator_ringmaster_wrapper: Plays some eval games between two players
+
+players are rows in the cloud bigtable (normally models_for_eval)
+each row is a unique pair of (binary, flags) along with name.
+
+The sh script has already:
+1. Downloaded both players (binaries and models)
+
+This script is responsible for:
+2. Setup ringmaster control file
+3. Call ringmaster
+4. Record results into CBT
+    4a. Rename games and upload to GCS
+
+"""
+
+import sys
+sys.path.insert(0, "..")
+
+import os
+import shutil
+import subprocess
+import time
+from collections import namedtuple
+
+from google.cloud import bigtable
+from tensorflow import gfile
+
+# Each import must be added to the dockerfile (via Makefile)
+import bigtable_output
+from bigtable_input import METADATA
+
+
+CTL_NAME = "ringmaster_evals"
+CTL_FILENAME = CTL_NAME + ".ctl"
+CTL_GAME_DIR = CTL_NAME + ".games"
+CTL_LOG = CTL_NAME + ".log"
+CTL_REPORT = CTL_NAME + ".report"
+
+MODEL_ROW_FMT = "m_eval_{}"
+
+CTL_FILE = '''
+competition_type = "playoff"
+description = " Testing models_for_eval "
+board_size = 19
+komi = 7.5
+
+record_games = True
+stderr_to_log = True
+
+def MinigoPlayer(path, model_pb, flags):
+    return Player(
+        "bin/bazel-bin/cc/gtp --model='tf,{{}}' {{}}".format(model_pb, flags),
+        cwd=path,
+        environ={{"LD_LIBRARY_PATH": "bin/cc/tensorflow/lib"}},
+        sgf_player_name_from_gtp=False)
+
+p_a = "{m_a.name}_{m_a.hash}"
+p_b = "{m_b.name}_{m_b.hash}"
+players = {{
+    p_a: MinigoPlayer("{m_a.hash}", "{m_a.model_pb}", "{m_a.flags}"),
+    p_b: MinigoPlayer("{m_b.hash}", "{m_b.model_pb}", "{m_b.flags}"),
+}}
+
+matchups = [
+    Matchup(p_a, p_b, id="{m_a.hash}_vs_{m_b.hash}_t_{start_time}",
+            alternating=True, number_of_games={num_games})
+]
+'''
+
+Model = namedtuple("Model", ["hash", "name", "model_pb", "flags"])
+
+def setup_ringmaster(model_a, model_b, start_time, num_games):
+    with open(CTL_FILENAME, "w") as ctl_f:
+        ctl_f.write(CTL_FILE.format(
+            m_a=model_a,
+            m_b=model_b,
+            start_time=start_time,
+            num_games=num_games))
+
+
+def call_ringmaster(num_games):
+    process = subprocess.run(
+        ["ringmaster", CTL_FILENAME, "run"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        timeout=num_games*20*60)
```

Comment:
this might be nice to have as a global or  a flag.

Reply:
There's a lovely todo in your name on line 212 :stuck_out_tongue: